contracts/cw2981-royalties/src/lib.rs

```diff
@@ -0,0 +1,255 @@
+pub mod msg;
+pub mod query;
+
+use cosmwasm_std::to_binary;
+pub use query::{check_royalties, query_royalties_info};
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::msg::Cw2981QueryMsg;
+use cosmwasm_std::Empty;
+use cw721_base::Cw721Contract;
+pub use cw721_base::{ContractError, InstantiateMsg, MintMsg, MinterResponse};
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug, Default)]
+pub struct Trait {
+    pub display_type: Option<String>,
+    pub trait_type: String,
+    pub value: String,
+}
+
+// see: https://docs.opensea.io/docs/metadata-standards
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug, Default)]
+pub struct Metadata {
+    pub image: Option<String>,
+    pub image_data: Option<String>,
+    pub external_url: Option<String>,
+    pub description: Option<String>,
+    pub name: Option<String>,
+    pub attributes: Option<Vec<Trait>>,
+    pub background_color: Option<String>,
+    pub animation_url: Option<String>,
+    pub youtube_url: Option<String>,
+    /// specify whether royalties are set on this token
+    pub royalty_payments: bool,
+    /// This is how much the minter takes as a cut when sold
+    pub royalty_percentage: Option<u64>,
```

Comment:
Question: Why do we need the `bool` above if this is an option?

I figure `royalty_payments = royalty_percentage.is_some()`
Also, only adding optional fields makes it more compatible with other contracts (they just happen to have it unset)

Reply:
Good point. I guess as long as the query handler returns the result of that lookup, then the interface remains the same. Will change.